### PR TITLE
FromIterator and Extend for bitsets

### PR DESCRIPTION
**Pull request by Aceeri**
This change is [Reviewable](https://reviewable.io/reviews/slide-rs/hibitset/23)

### CODE REVIEW

File: src/ops.rs [hunk 1]
```diff
@@ -200,6 +201,62 @@ operator!(impl<('a)(A, B)> for &'a BitSetOr<A, B>);
...
+                bitset
+            }
+        }
+
+        impl Extend<$index> for $bitset {
+            fn extend<T>(&mut self, iter: T)
+            where
+                T: IntoIterator<Item = $index>,
+            {
+                for item in iter.into_iter() {
```
Comment (torkleyy) [inline src/ops.rs]: There's no need to call `into_iter`, is there?

File: src/ops.rs [hunk 2]
```diff
@@ -200,6 +201,62 @@ operator!(impl<('a)(A, B)> for &'a BitSetOr<A, B>);
 operator!(impl<()(A, B)> for BitSetXor<A, B>);
 operator!(impl<('a)(A, B)> for &'a BitSetXor<A, B>);
 
+macro_rules! iterator {
+    ( $index:ty, $bitset:ident ) => {
+        impl FromIterator<$index> for $bitset {
```
Comment (torkleyy) [inline src/ops.rs]: Why is this macro generic over the index?

**Review by torkleyy — APPROVED** [review]

File: src/ops.rs (hunk 2 above)
Comment (Aceeri) [reply src/ops.rs]: Remnants from when I thought it would be easier to specify &'a Index and Index for the traits.

File: src/ops.rs (hunk 1 above)
Comment (Aceeri) [reply src/ops.rs]: Nope

**Review by torkleyy — APPROVED** [review]

**Comment by torkleyy** [comment]
Thank you!

bors r+

**Pull request merged**